compiler/src/codegen/compcore.re

```diff
@@ -90,6 +81,7 @@ let console_mod = "console";
 let tracepoint_ident = Ident.create_persistent("tracepoint");
 
 let grain_main = "_gmain";
+let grain_type_metadata = "_gtype_metadata";
```

Comment:
Type metadata is now explicitly provided by a special type metadata function in the module. This allows modules to choose to define only types but still have type information loaded.